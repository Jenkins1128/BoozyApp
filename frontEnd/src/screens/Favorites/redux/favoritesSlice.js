import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';
const initialState = {
	currentState: {
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: []
	}
};

export const getFavoritesAsync = createAsyncThunk('getFavoritesAsync/status', async (data, { rejectWithValue }) => {
	try {
		const response = await axios.get(`${baseUrl}/get`);
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getFavoritesSlice = createSlice({
	name: 'getFavoritesAsync',
	initialState,
	reducers: {
		resetGetFavoritesRequestStatus: (state) => {
			state.currentState.getFavoritesRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getFavoritesAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				getFavoritesRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getFavoritesAsync.rejected, (state) => {
			state.currentState.getFavoritesRequestStatus = 'rejected';
		});
	}
});

export const { resetGetFavoritesRequestStatus } = getFavoritesSlice.actions;
export const selectFavoritesState = (state) => state.profile.currentState;
export default getFavoritesSlice.reducer;
