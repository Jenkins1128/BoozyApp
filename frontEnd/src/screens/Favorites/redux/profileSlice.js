import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: []
	}
};

export const getFavoritesAsync = createAsyncThunk('getFavoritesAsync/status', async (data, { rejectWithValue }) => {
	try {
		const response = await axios.get('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/get');
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getFavoritesSlice = createSlice({
	name: 'getFavoritesAsync',
	initialState,
	reducers: {
		resetGetFavoritesRequestStatus: (state) => {
			state.currentState.getFavoritesRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getFavoritesAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				getFavoritesRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getFavoritesAsync.rejected, (state) => {
			state.currentState.getFavoritesRequestStatus = 'rejected';
		});
	}
});

export const { resetGetFavoritesRequestStatus } = getFavoritesSlice.actions;
export const selectFavoritesState = (state) => state.profile.currentState;
export default getFavoritesSlice.reducer;
