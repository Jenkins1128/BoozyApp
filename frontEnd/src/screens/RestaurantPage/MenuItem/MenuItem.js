import React from 'react';
import { StyleSheet, Text, View, TouchableWithoutFeedback } from 'react-native';

const MenuItem = ({ keyval, val, dismiss }) => {
	return (
		<TouchableWithoutFeedback onPress={dismiss}>
			<View key={keyval} style={styles.item}>
				<Text style={styles.priceText}>${val.price}</Text>
				<Text style={styles.descText}>{val.content}</Text>
			</View>
		</TouchableWithoutFeedback>
	);
};

export default MenuItem;

const styles = StyleSheet.create({
	item: {
		padding: 20,
		paddingRight: 100,
		borderBottomWidth: 2,
		borderBottomColor: '#ededed'
	},
	priceText: {
		color: 'black'
	},
	descText: {
		color: 'black'
	}
});
