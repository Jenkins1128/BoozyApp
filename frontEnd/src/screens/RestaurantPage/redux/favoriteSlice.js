import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';

const initialState = {
	currentState: {}
};

export const favoriteAsync = createAsyncThunk('favoriteAsync/status', async (data, { rejectWithValue }) => {
	const { restaurantId, name } = data;
	try {
		//favorite restaurant
		await axios.post(`${baseUrl}/user/add`, {
			restaurantId: restaurantId,
			name: name
		});
		//add restaurants to your favorites
		const response = await axios.post(`${baseUrl}/favorites`, {
			restaurantId: restaurantId,
			name: name
		});
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

const favoriteSlice = createSlice({
	name: 'favoriteAsync',
	initialState,
	reducers: {
		resetFavorite: (state) => {
			state.currentState.favoriteRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(favoriteAsync.fulfilled, (state, { payload }) => {
				Object.assign(state.currentState, {
					favoriteColor: payload[0]['contains'] ? 'red' : 'white',
					favoriteRequestStatus: 'fulfilled'
				});
			})
			.addCase(favoriteAsync.rejected, (state) => {
				state.currentState.favoriteRequestStatus = 'rejected';
			});
	}
});

export const { resetFavorite } = favoriteSlice.actions;

export const selectFavoriteState = (state) => state.favorite.currentState;
export default favoriteSlice.reducer;
