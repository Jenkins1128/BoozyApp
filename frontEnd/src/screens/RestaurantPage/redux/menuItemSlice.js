import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {}
};

export const postMenuItemAsync = createAsyncThunk('postMenuItemAsync/status', async (data, { rejectWithValue }) => {
	const { description, price, restaurantName, restaurantId } = data;
	try {
		const response = await axios.post('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/user/' + restaurantId + '/menu', {
			content: description,
			price: price,
			name: restaurantName
		});
		return response.status;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

const postMenuItemSlice = createSlice({
	name: 'postMenuItemAsync',
	initialState,
	reducers: {
		updatePrice: (state, { payload }) => {
			state.currentState = { ...state.currentState, price: payload.price };
		},
		updateDescription: (state, { payload }) => {
			state.currentState = { ...state.currentState, description: payload.description };
		},
		resetMenuItem: (state) => {
			state.currentState = { ...state.currentState, price: '', description: '', postMenuItemRequestStatus: 'idle' };
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(postMenuItemAsync.fulfilled, (state) => {
				state.currentState = { ...state.currentState, postMenuItemRequestStatus: 'fulfilled' };
			})
			.addCase(postMenuItemAsync.rejected, (state) => {
				state.currentState = { ...state.currentState, postMenuItemRequestStatus: 'rejected' };
			});
	}
});

export const { updatePrice, updateDescription, resetMenuItem } = postMenuItemSlice.actions;

export const selectMenuItemState = (state) => state.menuItem.currentState;
export default postMenuItemSlice.reducer;
