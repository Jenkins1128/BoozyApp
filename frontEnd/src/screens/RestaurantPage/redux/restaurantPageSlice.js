import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl, defaultRestaurantImageUrl } from '../../../helpers/constants';

const initialState = {
	currentState: {
		restaurantId: '',
		restaurantName: '',
		restaurantImage: defaultRestaurantImageUrl,
		phoneNumber: '',
		allCategories: '',
		rating: 0.0,
		reviews: 0,
		favoriteColor: 'white',
		changeFavoritedColor: '',
		starCount: 0,
		favorited: false,
		rated: false,
		price: 0.0,
		description: '',
		menuItemArray: []
	}
};

export const getMenuItemsAsync = createAsyncThunk('getMenuItemsAsync/status', async (restaurantId, { rejectWithValue }) => {
	try {
		const response = await axios.post(`${baseUrl}/menu/list`, {
			restaurantId: restaurantId
		});
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const restaurantPageSlice = createSlice({
	name: 'getMenuItemsAsync',
	initialState,
	reducers: {
		updateState: (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantId: payload.params.id,
				restaurantName: payload.params.name,
				restaurantImage: payload.params.img,
				phoneNumber: payload.params.phone,
				allCategories: payload.params.allCategories,
				rating: payload.params.rating,
				reviews: payload.params.reviews,
				rated: payload.params.alreadyRated,
				favorited: payload.params.alreadyFavorited,
				starCount: 0
			});
		},
		updateFavoriteColor: (state, { payload }) => {
			state.currentState.favoriteColor = payload.color;
		},
		updateStarCount: (state, { payload }) => {
			state.currentState.starCount = payload.starCount;
		},
		updateMenuItems: (state, { payload }) => {
			state.currentState.menuItemArray.push(payload.menuItem);
		},
		resetRestaurantPageRequestStatus: (state) => {
			state.currentState.restaurantPageRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(getMenuItemsAsync.fulfilled, (state, { payload }) => {
				Object.assign(state.currentState, {
					restaurantPageRequestStatus: 'fulfilled',
					menuItemArray: payload
				});
			})
			.addCase(getMenuItemsAsync.rejected, (state) => {
				state.currentState.restaurantPageRequestStatus = 'rejected';
			});
	}
});

export const { updateState, updateFavoriteColor, updateStarCount, updateMenuItems, resetRestaurantPageRequestStatus } = restaurantPageSlice.actions;

export const selectRestaurantPageState = (state) => state.restaurantPage.currentState;
export default restaurantPageSlice.reducer;
