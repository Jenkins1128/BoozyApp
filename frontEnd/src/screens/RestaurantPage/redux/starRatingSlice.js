import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {}
};

export const starRatingAsync = createAsyncThunk('starRatingAsync/status', async (data, { rejectWithValue }) => {
	const { rating, restaurantId, restaurantName } = data;
	try {
		const response = await axios.post('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/user/' + restaurantId + '/restaurants', {
			content: rating,
			name: restaurantName
		});
		return response.status;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const starRatingSlice = createSlice({
	name: 'starRatingAsync',
	initialState,
	reducers: {
		resetStarRatingRequestStatus: (state) => {
			state.currentState.starRatingRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(starRatingAsync.fulfilled, (state) => {
				state.currentState.starRatingRequestStatus = 'fulfilled';
			})
			.addCase(starRatingAsync.rejected, (state) => {
				state.currentState.starRatingRequestStatus = 'rejected';
			});
	}
});

export const { resetStarRatingRequestStatus } = starRatingSlice.actions;

export const selectStarRatingState = (state) => state.starRating.currentState;
export default starRatingSlice.reducer;
