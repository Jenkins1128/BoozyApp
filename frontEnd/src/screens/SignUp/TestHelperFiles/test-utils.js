import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const INITIAL_STATE = {
	signup: {
		currentState: {
			requestStatus: 'idle',
			isLoading: false,
			email: 'xxx@gmail.com',
			password: 'xxx'
		}
	},
	login: {
		currentState: {
			requestStatus: 'idle',
			isLoading: false,
			email: 'xxx@gmail.com',
			password: 'xxx'
		}
	}
}; // Or use value from root reducer
const mockStore = configureStore([]);
const store = mockStore(INITIAL_STATE);

const reduxRender = (ui, options) =>
	render(ui, {
		wrapper: ({ children }) => <Provider store={store}>{children}</Provider>,
		...options
	});

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { reduxRender, store };
