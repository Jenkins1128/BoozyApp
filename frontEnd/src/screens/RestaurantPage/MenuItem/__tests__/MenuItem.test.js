import React from 'react';
import { render, cleanup, fireEvent } from '../../../../TestHelperFiles/redux/test-utils';
import MenuItem from '../MenuItem';

describe('<MenuItem />', () => {
	let val;
	let dismiss;
	let rendered;

	beforeEach(() => {
		val = {
			price: 6,
			content: 'test'
		};
		dismiss = jest.fn();
		rendered = render(<MenuItem val={val} dismiss={dismiss} />);
	});

	afterEach(cleanup);

	//snapshot
	it('should render correctly', () => {
		expect(rendered).toMatchSnapshot();
	});

	//content
	it('should render item styles correctly', () => {
		const itemViewComponent = rendered.getByTestId('container');
		const given = itemViewComponent.props.style;
		const result = {
			padding: 20,
			paddingRight: 100,
			borderBottomWidth: 2,
			borderBottomColor: '#ededed'
		};
		expect(result).toMatchObject(given);
	});

	it('should render priceText styles correctly', () => {
		const priceTextComponent = rendered.getByTestId('priceText');
		const given = priceTextComponent.props.style;
		const result = {
			color: 'black'
		};
		expect(result).toMatchObject(given);
	});

	it('should render descText styles correctly', () => {
		const descTextComponent = rendered.getByTestId('descText');
		const given = descTextComponent.props.style;
		const result = {
			color: 'black'
		};
		expect(result).toMatchObject(given);
	});

	//interaction
	it('should fire dismiss button event on TouchableWithoutFeedback', () => {
		const dismissButton = rendered.getByTestId('container');
		fireEvent.press(dismissButton);
		expect(dismiss).toHaveBeenCalledTimes(1);
	});
});
