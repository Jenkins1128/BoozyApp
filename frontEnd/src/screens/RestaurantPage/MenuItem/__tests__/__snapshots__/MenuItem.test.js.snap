// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuItem /> should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomColor": "#ededed",
      "borderBottomWidth": 2,
      "padding": 20,
      "paddingRight": 100,
    }
  }
  testID="container"
>
  <Text
    style={
      Object {
        "color": "black",
      }
    }
    testID="priceText"
  >
    $
    6
  </Text>
  <Text
    style={
      Object {
        "color": "black",
      }
    }
    testID="descText"
  >
    test
  </Text>
</View>
`;
